### PR TITLE
Update internal documentation about "How to make a release".

### LINKED ISSUE #4999: sdist 0.8.2 contains Theano.egg-info with bad permissions
Appears the sdist for the `0.8.2` release has a `Theano.egg-info` directory with bad permissions. Below is a listing that shows this directory and its contents.

``` bash
$ ls -l Theano-0.8.2/Theano.egg-info 
total 96
-rw-r-----@ 1 kirkhamj  1899195968  11440 Apr 21 08:04 PKG-INFO
-rw-r-----@ 1 kirkhamj  1899195968  19445 Apr 21 08:04 SOURCES.txt
-rw-r-----@ 1 kirkhamj  1899195968      1 Apr 21 08:04 dependency_links.txt
-rw-r-----@ 1 kirkhamj  1899195968     47 Apr 21 08:04 pbr.json
-rw-r-----@ 1 kirkhamj  1899195968    112 Apr 21 08:04 requires.txt
-rw-r-----@ 1 kirkhamj  1899195968      7 Apr 21 08:04 top_level.txt
```

At this point, we would like to know if we can simply remove this directory. Though it may be worthwhile having a `post` release that corrects this issue in some fashion.

cc @0x0L @davidslac

### PR DESCRIPTION
**Pull request by notoraptor**
fix #4999 and other typos.

@nouiz @lamblin

### CODE REVIEW

File: doc/internal/how_to_release.txt [hunk 1]
```diff
@@ -64,11 +64,20 @@ The documentation will be automatically regenerated in the next few hours.
 Generate and upload the package
 ===============================
 
-For release candidates, only upload on PyPI.
+For non-final releases (e.g., release candidates, alpha, beta), upload on PyPI only.
```
Comment (nouiz) [inline doc/internal/how_to_release.txt]: For release candidate, we should also email theano-user and theano-dev. For the final, we what other venue + theano-announce.

Comment (nouiz) [inline doc/internal/how_to_release.txt]: the "only upload to PyPI" isn't clear if this is for the rest of the document of this section alone.

Can you remove that sentenses and put in the "on mloss.org" section something like:
To this only for final relelase.

Comment (nouiz) [inline doc/internal/how_to_release.txt]: I'll kill travis on this commit.

Comment (notoraptor) [inline doc/internal/how_to_release.txt]: Done !

**Comment by notoraptor** [comment]
Another update to specify email lists for alpha and beta releases.

This is the current email rules:

> For final releases, send the e-mail to the following mailing lists:
> 
> * theano-users
> * theano-announce
> * numpy-discussion@scipy.org
> * scipy-user@scipy.org
> * G+, Scientific Python: https://plus.google.com/communities/108773711053400791849
> 
> For release candidates, only e-mail:
> 
> * theano-announce
> * theano-dev
> * theano-users
> 
> For alpha and beta releases, only e-mail:
> 
> * theano-dev
> * theano-users

(I have not added `theano-announce` list for alpha/beta releases)

**Comment by lamblin** [comment]
One check in Travis timed out, and it was not related to documentation, so I'm merging.

**Pull request merged**